pkg/gateway/gateway.go

```diff
@@ -520,7 +520,7 @@ func (n *jfsObjects) GetObjectInfo(ctx context.Context, bucket, object string, o
 		return
 	}
 	if strings.HasSuffix(object, sep) && !fi.IsDir() {
-		err = jfsToObjectErr(ctx, os.ErrNotExist, bucket, object)
```

Comment:
Will this change any behavior?

Reply:
os.ErrNotExist will give 500 http code InternalError, syscall.ENOENT give 404 code. i think 404 is better

Reply:
I see, can you send another PR for this one?

Reply:
ok, i sent a new PR: https://github.com/juicedata/juicefs/pull/1158